app/Libraries/ChatFilters.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Libraries;
+
+use App\Models\ChatFilter;
+
+class ChatFilters
+{
+    private $chatFilters;
+
+    public function all()
+    {
+        if (!isset($this->chatFilters)) {
+            $this->fetch();
+        }
+
+        return $this->chatFilters;
+    }
+
+    public function fetch()
+    {
+        $localCacheVersion = cache()->get('chat_filters_local_cache_version');
+        $localStorage = config('cache.local');
```

Comment:
this is on my todo list for the groups one but this process can be slightly improved:

- set the entire cache repository to variable instead of just the name
  - `$localCache = cache()->store(config('cache.local'))`
- use static cache key with version as part of its value so cache directory doesn't get filled up with outdated cache data
  - `$localCache->forever('chat_filters', ['version' => $localCacheVersion, 'data' => ChatFilter::all()])`